src/smartfin-fw3.ino

```diff
@@ -1,20 +1,118 @@
 /*
  * Project smartfin-fw3
- * Description:
+ * Description: 
  * Author:
  * Date:
  */
 #include "Particle.h"
+#include "states.hpp"
+#include "task.hpp"
+#include "product.hpp"
+#include "cli/cli.hpp"
+#include "cli/conio.hpp"
+#include "cli/flog.hpp"
+
+
+// Statemachine for handeling task-switching
+typedef struct StateMachine_
+{
+    STATES_e state;
+    Task* task;
+}StateMachine_t;
+
+static CLI cliTask;
+
+
+// Holds the list of states and coresponding tasks
+static StateMachine_t stateMachine[] = 
+{
+    {STATE_CLI, &cliTask},
+    {STATE_NULL, NULL}
+};
+
+static STATES_e currentState;
+
+static StateMachine_t* findState(STATES_e state);
+static void initalizeTaskObjects(void);
+void mainThread(void* args);
+void printState(STATES_e state);
 
 // setup() runs once, when the device is first turned on.
 void setup() {
-    // Put initialization like pinMode and begin functions here.
-    Serial.begin(9500);
+    Serial.begin(SF_SERIAL_SPEED);
+
+    currentState = STATE_CLI;
+
+    
+    FLOG_Initialize();
+    FLOG_AddError(FLOG_SYS_START, 0); 
+    time32_t bootTime = Time.now();
+    SF_OSAL_printf("Boot time: ", bootTime);
```

Comment:
Note that this will only output `Boot Time: `!  This should also probably have a newline after.